### ISSUE TITLE
template v-for key conflict

### ISSUE THREAD

**Issue by Ricky816845** [vuejs/eslint-plugin-vue#1940]
eslint vertion:6.8.0.
eslint-plugin-vue version:6.2.2
node version 16.15.1
system win11

config in package.json
``` json5
 "eslintConfig": {
        "root": true,
        "env": {
            "node": true
        },
        "extends": [
            "plugin:vue/essential",
            "eslint:recommended"
        ],
        "parserOptions": {
            "parser": "babel-eslint"
        },
        "rules": {}
    },
```

I have template v-for element (solution from vuetify) 
```vue
        <v-list-item-group>
          <template v-for="(org, index) in OrgList">
            <v-list-item :key="org.cid">
            some content
            </v-list-item>
            <v-divider :key="index"></v-divider>
          </template>
        </v-list-item-group>
```

**What did you expect to happen?**
no error expected  whether I put ":key" on "v-list-item" or template 

**What actually happened?**
![image](https://user-images.githubusercontent.com/20158450/181142934-67de2a3c-49b7-4c4e-8225-f6bb5c25b4b8.png)
![image](https://user-images.githubusercontent.com/20158450/181143246-577e5951-317e-4be3-b347-46c2bc910c9a.png)

demo :
https://github.com/Ricky816845/template-v-for-key-problem

**Comment by ota-meshi** [comment]
Thank you for posting issue.

I think the rules are working correctly. Perhaps Volra's warning is wrong.
